tests/test_git_colocated.rs

```diff
@@ -357,6 +357,70 @@ fn test_git_colocated_squash_push_undo() {
     "###);
 }
 
+// TODO: Fix the BUG this reveals. It may be sufficient to make `jj git import`
+// use the "last seen git refs" file.
```

Comment:
Doesn't `jj git import` already use that file since one of the commits in this PR?

Reply:
No, I need to elaborate on this comment. As I currently implemented it, jj git import writes the last seen git refs file, but does not read it. It still uses the git refs as the base for the branch conflict resolution.

Reply:
> As I currently implemented it, jj git import writes the last seen git refs file, but does not use the information it reads from that file. It still uses the git refs as the base for the branch conflict resolution.

Is that how it should work or just how it currently works? I feel like we should use the last seen state when importing refs.

Reply:
Yes, but I haven't worked on that yet.

Reply:
I see. I just wanted to check that we agreed on how it should work. It seems like this PR already improves on the current state, so feel free to leave that for later.

Reply:
This and locking are next on my list. I wanted to think about how to have a test or tests that can tell the difference for this as well. I'm not 100% sure if either of the "bug demo" tests I have here would.

Reply:
I removed this comment and clarified what needs to happen in `git.rs` next to the import function.

Aside: Locking turned out to be easy, you've already done all the work.